src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -81,27 +82,38 @@ public static boolean isXPStyleEnabled() {
     }
 
     // this should be called only with writeLock held
-    private static Long getThemeImpl(String widget) {
-        Long theme = widgetToTheme.get(widget);
+    private static Long getThemeImpl(String widget, int dpi) {
+        Long theme = null;
+
+        if(dpiAwareWidgetToTheme.containsKey(dpi)) {
```

Comment:
Can we just call `dpiAwareWidgetToTheme.get(dpi)` and compare result with `null` ? It's faster then 2 containsKey+get calls.